hearth/lib/hearth/auth_schemes.rb

```diff
@@ -0,0 +1,29 @@
+# frozen_string_literal: true
+
+module Hearth
+  module AuthSchemes
+    # Base class for all AuthScheme classes.
+    class Base
+      def initialize(scheme_id:)
```

Comment:
Why not also take identity_type and signer as parameters here?

Reply:
Signer is no longer a thing that needs to be configurable - the signing properties are used to modify signing behavior instead. Identity type will always be static. I guess both can be passed in here, but it was not called out.